### PR TITLE
Add "add icon" step to windows pipeline

### LINKED ISSUE #585: Add icon to ribasim.exe
Can be done using https://github.com/electron/rcedit

```
$ rcedit "path-to-exe-or-dll" --set-icon "path-to-ico"
```

Probably needs a .ico, right now we only have a .png: https://github.com/Deltares/Ribasim/blob/v2023.08.0/qgis/icon.png
EDIT: see favicon.ico in https://github.com/Deltares/Ribasim-NL

This would be an improvement over
image

### PR DESCRIPTION
**Pull request by Jingru923**
Fixes #585 

It is tested on local machine. After ribasim binary is built, if you run `pixi run add-ribasim-icon`, the ribasim.exe will have an ribasim icon. The image file is checked in utils/ folder. This "add icon" step is added to the windows build pipeline, it is right after the build step

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -104,6 +104,7 @@ codegen = { cmd = "julia --project utils/gen_python.jl && ruff format python/rib
     "python/ribasim/ribasim/schemas.py",
 ] }
 # Publish
+add-ribasim-icon ={ cmd = "rcedit \"build/ribasim/ribasim.exe\" --set-icon \"utils/ribasimicon.ico\""}
```
Comment (visr) [inline pixi.toml:107]:
```suggestion
add-ribasim-icon ={ cmd = "rcedit build/ribasim/ribasim.exe --set-icon utils/ribasimicon.ico"}
```

File: utils/ribasimicon.ico
Comment (visr) [inline utils/ribasimicon.ico:1]: Would be better in `docs\assets\ribasim.ico`

**Review by visr — COMMENTED** [review]
Two nitpicks, looks good to me otherwise if the pixi.lock diff is based on main.

**Comment by Jingru923** [comment]
I downloaded one of the windows build artifacts from TC, it worked: :D
image

**Review by visr — APPROVED** [review]

**Pull request merged**